src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -1920,29 +1919,38 @@ public void Dispose()
         public Dictionary<string, HashSet<string>> GetDisabledSubscribersCollections(HashSet<string> tombstoneCollections)
         {
             var dict = new Dictionary<string, HashSet<string>>(StringComparer.OrdinalIgnoreCase);
-            foreach (var replicationDestination in Destinations)
+            foreach (var replicationDestination in Destinations.Where(x => x.Disabled))
             {
-                if (replicationDestination.Disabled)
-                    dict[replicationDestination.FromString()] = tombstoneCollections;
+                dict[replicationDestination.GetDefaultTaskName()] = tombstoneCollections;
             }
 
             using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
             using (ctx.OpenReadTransaction())
             {
                 var externals = _server.Cluster?.ReadRawDatabaseRecord(ctx, Database.Name)?.ExternalReplications;
+                var hubPullReplications = _server.Cluster?.ReadRawDatabaseRecord(ctx, Database.Name)?.GetHubPullReplications();
```

Comment:
you are reading record 3x here, we can read it once

Reply:
Done.